src/main/java/book/store/service/impl/CartItemServiceImpl.java

```diff
@@ -46,9 +47,14 @@ public CartItemResponseDto getById(Long id) {
 
     @Override
     public void update(Long cartItemId, CartItemUpdateRequestDto requestDto) {
-        CartItem cartItem = cartItemRepository.getReferenceById(cartItemId);
-        cartItem.setQuantity(requestDto.getQuantity());
-        cartItemRepository.save(cartItem);
+        CartItem cartItem = cartItemRepository.findById(cartItemId).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find cart item with id " + cartItemId));
+        if (requestDto.getQuantity() == 0) {
+            cartItemRepository.deleteById(cartItemId);
+        } else {
+            cartItem.setQuantity(requestDto.getQuantity());
+            cartItemRepository.save(cartItem);
+        }
```

Comment:
useless logic, 
What is the reason that you did it?
If it is after the tests, it is better to rewrite the tests than to change the basic logic
```suggestion
            cartItem.setQuantity(requestDto.getQuantity());
            cartItemRepository.save(cartItem);
```